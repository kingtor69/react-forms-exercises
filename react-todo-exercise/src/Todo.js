import React from 'react';

const Todo = ({ id, todo, deleteTodo }) => {
  const htmlId = `todo${id}`;
  const testId = `test-${htmlId}`;
  const deleteThisTodo = () => { deleteTodo(id) };
  return (
    <li className="Todo">
      <button 
        className="deleteTodoButton"
        onClick={deleteThisTodo}
        id={htmlId}
      >
        X
      </button>
      {todo}
    </li>
  );
};

export default Todo;
