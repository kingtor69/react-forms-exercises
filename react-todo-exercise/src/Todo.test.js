import { render } from '@testing-library/react';
import Todo from './Todo';

it('renders without crashing', () => {
  render(<Todo />);
});

it('renders snapshot', () => {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});
