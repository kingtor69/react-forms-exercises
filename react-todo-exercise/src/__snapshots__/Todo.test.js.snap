// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<DocumentFragment>
  <li
    class="Todo"
  >
    <button
      class="deleteTodoButton"
      id="todoundefined"
    >
      X
    </button>
  </li>
</DocumentFragment>
`;
